incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -284,14 +292,38 @@ func (r *ObjectReconciler) onDelete(ctx context.Context, log logr.Logger, nnm ty
 	return nil
 }
 
+func (r *ObjectReconciler) ensureCanPropagate(nsNm string, inst *unstructured.Unstructured) bool {
+	r.Forest.Lock()
+	defer r.Forest.Unlock()
+	ns := r.Forest.Get(nsNm)
+	if !ns.Exists() {
+		return false
+	}
+	//TODO TAYLOR
+	var conditionString string
+	if len(strings.Split(r.GVK.String(), ".")) > 1 {
+		affected := strings.Split(r.GVK.String(), ".")
+		conditionString = fmt.Sprint(affected[0], "/", affected[1], "/", affected[2], "/", nsNm, "/", ns.Name())
```

Comment:
The last piece in the string should be ```inst.GetName()``` since it's the name of the object not namespace name. ```ns.Name()``` is the same as ```nsNm``` here.